tests/testthat/test-data-frame.R

```diff
@@ -220,39 +220,44 @@ test_that("Can convert named atomic vectors to data frame", {
 })
 
 
-test_that("as_tibble() validates by default (#278)", {
+test_that("as_tibble() checks names by default (#278)", {
   df <- tibble(a = 1, b = 2)
   names(df) <- c("", NA)
   expect_error(
     as_tibble(df),
     error_column_must_be_named(1:2),
     fixed = TRUE
   )
-
-  expect_error(as_tibble(df, .tidy_names = TRUE), NA)
```

Comment:
Do we test conversion of NA and empty names anywhere?

Reply:
Honestly, I just did minimal patch up of existing tests. But I can do a more comprehensive review of the test coverage of name repair.

Reply:
I've worked quite a bit on the tests now. I read / rationalized all existing tests around names and added new ones for the new functionality and helpers.